pkg/pillar/cmd/zedagent/handlecontent.go

```diff
@@ -60,6 +75,7 @@ func parseContentInfoConfig(ctx *getconfigContext,
 		contentConfig := new(types.ContentTreeConfig)
 		contentConfig.ContentID, _ = uuid.FromString(cfgContentTree.GetUuid())
 		contentConfig.DatastoreID, _ = uuid.FromString(cfgContentTree.GetDsId())
+		contentConfig.FallbackDatastoreIDsList, _ = stringsToUuids(cfgContentTree.GetFallbackDsIdsList())
```

Comment:
Sorry, I did not get the comment. The intention is to pass the list to the final "destination", the downloader. Probably there is simpler approach, which I don't see.

Reply:
I mean, should we store DatastoreID and FallbackDatastoreIDsList as separate objects in our internal types (using in API is enforced)? It sounds reasonable if one day we will have another config knob, e.g. `not use fallback datastores` or `enable tracing for fallback datastores`. Can you see any blocker of merging them into DatastoreIDs in zedagent?

Reply:
The only blocker is that there is a bunch of code, which accesses the `DatastoreID` member expecting one datastore, not a list. Unfortunately the whole patchset is a hack, which handles the datastore list at the bottom (downloader) of the whole stack, and the rest of the code works as before - handling datastore as a singleton,  eventhough the main datastore has failed and we already fallbacked to one of the entries in the fallback list.

Of course I can do the following everywhere in the code: `s/DatastoreID/DatastoreIDs[0]/g`, but logically it will be the same, so I keep two members to emphasize that there is a "main datastore" and there is a "fallback list". 

Probably the proper way would be to have a list and a pointer (index) of a current "datastore" entry in the list. If one of the datastores in the list fails - a pointer (index) moves to the next entry in the list. But! Here I'm not quite confident how the rest of the code will behave if the "DatastoreID" suddenly changes underneath (in the `downloader`), I expect structures can be lost in the lists or mappings, which use `DatastoreID` as a key.

So here I'm very conservative and don't want to break the old code.

Reply:
But aren't most of that other code passthrough of DatastoreID thus it is easy to make it all pass through DatastoreIDList instead?

That way the logic in zedagent and downloader is the only ones which need some real changes.

Reply:
There are couple of dependencies which I do not know how to resolve right now:
1. `AddOrRefcountDownloaderConfig()` (handledownloader.go) creates `idHash` using the `blob.DatastoreID` for the target file name. This happens before we actually start downloading. Probably we can build `idHash` later in the subscriber of the `DownloaderConfig`. 
2. A bunch of  places in volumemgr/handleresolve.go and the whole resolve logic, which I do not touch for now.
3. ResolveKey() uses the `status.DatastoreID` in the `contenttretype.go`, again related to async resolve logic.
4. A bunch of `logObject.CloneAndAddField("datastore-id")` calls, which of course can be replaced with the `DatastoreIDs[0]`, but not sure is that a good idea or not.

So all these things will take some time once I get everything sorted out, so this first iteration is rather conservative and uses the fallback list inside the `downloader`, keeping the rest of the logic the same.

Reply:
So the main problem is that `DatastoreID` as a singleton is used along the whole downloader stack, it is published in structures as a target file, printed in logs, participates in async completion callbacks (resolver), but if we switch to the list of `DatastoreIDs` we actually can't use the datastore id before we actually start downloading in the `downloader` (at the bottom of the whole stack). And in this PR is not solved right now, it is clumsy and ugly: we just use first datastore as before.

Reply:
@eriknordmark ^^^